hazelcast/src/main/java/com/hazelcast/internal/services/WanSupportingService.java

```diff
@@ -33,5 +33,5 @@
      *                        or should it return after the event has been dispatched to the
      *                        appropriate member
      */
-    void onReplicationEvent(WanEvent event, WanAcknowledgeType acknowledgeType);
+    void onReplicationEvent(InternalWanEvent event, WanAcknowledgeType acknowledgeType);
```

Comment:
From `WanSupportingService` javaDoc:
```
/**
 * An interface that can be implemented by SPI services to give them the
 * ability to listen to WAN replication events.
 */
```
is it ok to use `impl.InternalWanEvent`) here since it seems an implementation detail?

Reply:
The javadoc is unclear here. It refers to the "custom user-defined services SPI" aka NodeEngine which was removed in 4.0. The usage of the class is ok here then.